test/e2e/e2e_test.go

```diff
@@ -180,6 +182,169 @@ var _ = ginkgo.Describe("Kueue", func() {
+						Resource(corev1.ResourceCPU, "1").
+						Resource(corev1.ResourceMemory, "1Gi").
+						Obj(),
+				).
+				AdditionalChecks("check1").
+				Obj()
+			gomega.Expect(k8sClient.Create(ctx, clusterQueue)).Should(gomega.Succeed())
+			localQueue = testing.MakeLocalQueue("main", ns.Name).ClusterQueue("cluster-queue").Obj()
+			gomega.Expect(k8sClient.Create(ctx, localQueue)).Should(gomega.Succeed())
+		})
+		ginkgo.AfterEach(func() {
+			gomega.Expect(util.DeleteLocalQueue(ctx, k8sClient, localQueue)).Should(gomega.Succeed())
+			gomega.Expect(util.DeleteAllJobsInNamespace(ctx, k8sClient, ns)).Should(gomega.Succeed())
+			util.ExpectClusterQueueToBeDeleted(ctx, k8sClient, clusterQueue, true)
+			util.ExpectResourceFlavorToBeDeleted(ctx, k8sClient, onDemandRF, true)
+		})
+
+		ginkgo.It("Should unsuspend a job only after all checks are cleared", func() {
+			// Use a binary that ends.
+			sampleJob = (&testingjob.JobWrapper{Job: *sampleJob}).Image("gcr.io/k8s-staging-perf-tests/sleep:v0.0.3", []string{"5s"}).Obj()
+			gomega.Expect(k8sClient.Create(ctx, sampleJob)).Should(gomega.Succeed())
+
+			createdWorkload := &kueue.Workload{}
+			wlLookupKey := types.NamespacedName{Name: workloadjob.GetWorkloadNameForJob(jobKey.Name), Namespace: ns.Name}
+
+			ginkgo.By("verify the check is added to the workload", func() {
+				gomega.Eventually(func() map[string]string {
+					if err := k8sClient.Get(ctx, wlLookupKey, createdWorkload); err != nil {
+						return nil
+					}
+					return slices.ToMap(createdWorkload.Status.AdmissionChecks, func(i int) (string, string) { return createdWorkload.Status.AdmissionChecks[i].Type, "" })
+
+				}, util.LongTimeout, util.Interval).Should(gomega.BeComparableTo(map[string]string{"check1": ""}))
+			})
+
+			ginkgo.By("waiting for the workload to be assigned", func() {
+				gomega.Eventually(func() bool {
+					if err := k8sClient.Get(ctx, wlLookupKey, createdWorkload); err != nil {
+						return false
+					}
+					return apimeta.IsStatusConditionTrue(createdWorkload.Status.Conditions, kueue.WorkloadAdmitted)
+
+				}, util.Timeout, util.Interval).Should(gomega.BeTrue())
+			})
+
+			ginkgo.By("checking the job remains suspended", func() {
+				createdJob := &batchv1.Job{}
+				jobKey := client.ObjectKeyFromObject(sampleJob)
+				gomega.Consistently(func() bool {
+					if err := k8sClient.Get(ctx, jobKey, createdJob); err != nil {
+						return false
+					}
+					return ptr.Deref(createdJob.Spec.Suspend, false)
+
+				}, util.ConsistentDuration, util.Interval).Should(gomega.BeTrue())
+			})
+
+			ginkgo.By("setting the check as successful", func() {
+				gomega.Eventually(func() error {
+					if err := k8sClient.Get(ctx, wlLookupKey, createdWorkload); err != nil {
+						return err
+					}
+					apimeta.SetStatusCondition(&createdWorkload.Status.AdmissionChecks, metav1.Condition{
+						Type:   "check1",
+						Status: metav1.ConditionTrue,
+						Reason: "ByTest",
+					})
```

Comment:
Let's do this with SSA. I expect clients to want to use SSA. (It might require force:true)